ProcessLib/ComponentTransport/ComponentTransportProcessData.h

```diff
@@ -0,0 +1,101 @@
+
+#include "PorousMediaProperties.h"
+
+namespace MeshLib
+{
+class Element;
+}
+
+namespace ProcessLib
+{
+template <typename ReturnType>
+struct Parameter;
+
+namespace ComponentTransport
+{
+struct ComponentTransportProcessData
+{
+    ComponentTransportProcessData(
+        PorousMediaProperties&& porous_media_properties_,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>&& viscosity_model_,
+        ProcessLib::Parameter<double> const& fluid_reference_density_,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>&& fluid_density_,
+        ProcessLib::Parameter<double> const& molecular_diffusion_coefficient_,
+        ProcessLib::Parameter<double> const& solute_dispersivity_longitudinal_,
+        ProcessLib::Parameter<double> const& solute_dispersivity_transverse_,
+        ProcessLib::Parameter<double> const& retardation_factor_,
+        ProcessLib::Parameter<double> const& decay_rate_,
+        Eigen::VectorXd const& specific_body_force_,
+        bool const has_gravity_)
+        : porous_media_properties(std::move(porous_media_properties_)),
+          viscosity_model(std::move(viscosity_model_)),
+          fluid_reference_density(fluid_reference_density_),
+          fluid_density(std::move(fluid_density_)),
+          molecular_diffusion_coefficient(molecular_diffusion_coefficient_),
+          solute_dispersivity_longitudinal(solute_dispersivity_longitudinal_),
+          solute_dispersivity_transverse(solute_dispersivity_transverse_),
+          retardation_factor(retardation_factor_),
+          decay_rate(decay_rate_),
+          specific_body_force(specific_body_force_),
+          has_gravity(has_gravity_)
+    {
+    }
+
+    ComponentTransportProcessData(ComponentTransportProcessData&& other)
+        : porous_media_properties(std::move(other.porous_media_properties)),
+          viscosity_model(other.viscosity_model.release()),
+          fluid_reference_density(other.fluid_reference_density),
+          fluid_density(other.fluid_density.release()),
+          molecular_diffusion_coefficient(
+              other.molecular_diffusion_coefficient),
+          solute_dispersivity_longitudinal(
+              other.solute_dispersivity_longitudinal),
+          solute_dispersivity_transverse(other.solute_dispersivity_transverse),
+          retardation_factor(other.retardation_factor),
+          decay_rate(other.decay_rate),
+          specific_body_force(other.specific_body_force),
+          has_gravity(other.has_gravity)
+    {
+    }
+
+    //! Copies are forbidden.
+    ComponentTransportProcessData(ComponentTransportProcessData const&) =
+        delete;
+
+    //! Assignments are not needed.
+    void operator=(ComponentTransportProcessData const&) = delete;
+
+    //! Assignments are not needed.
+    void operator=(ComponentTransportProcessData&&) = delete;
+
+    PorousMediaProperties porous_media_properties;
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty> viscosity_model;
```

Comment:
As the comment given about, you can replace
```
  std::unique_ptr<MaterialLib::Fluid::FluidProperty> viscosity_model;
  std::unique_ptr<MaterialLib::Fluid::FluidProperty> fluid_density;
```
with
```
  const std::unique_ptr<MaterialLib::Fluid::FluidProperties>
    _fluid_properties;
```
Here is an example to use FluidProperties:
https://github.com/ufz/ogs/blob/master/ProcessLib/LiquidFlow/LiquidFlowMaterialProperties.cpp#L75